astropy/coordinates/angles/_wrap.pyx

```diff
@@ -0,0 +1,55 @@
+#cython: language_level=3
+cimport cython
+cimport numpy as np
+from libc cimport math
+
+np.import_array()
+
+ctypedef fused dtype:
+    short
+    int
+    long
+    float
+    double
+
+
+@cython.ufunc
+@cython.cdivision(True)
+cdef dtype wrap_at(dtype angle, dtype wrap_angle, dtype period):
+    cdef dtype wrap_angle_floor
+    wrap_angle_floor = wrap_angle - period
+
+    if dtype is float or dtype is double:
+        if math.isnan(angle):
+            return angle
+
+    if (angle >= wrap_angle_floor) and (angle < wrap_angle):
+        return angle
+
+    cdef int wraps
+    wraps = <int> math.floor((angle - wrap_angle_floor) / period)
```

Comment:
Any reason not to use `(angle - wrap_angle_floor) // period` (as in the original code) ?

Reply:
I believe `//` involves more python-API overhead, while using `@cython.cdivision(True)` accomplishes the same thing at a lower cost. Is that right @maxnoe ?

Reply:
cython supports it (just calling `floor`, casting to int is still needed) : 
```
+11: @cython.cdivision(True)
 12: def div2(float angle, float period):
 13:     cdef int wraps
+14:     wraps = int(angle // period)
  __pyx_v_wraps = ((int)floor(__pyx_v_angle / __pyx_v_period));
+15:     return wraps
```

Reply:
What I pasted above is the output of `cythonize -a`, and I don't see Python interaction here.
It's as fast as your version (well it's just replacing with `floor`), just a bit more readable/pythonic, but `floor` is fine too ...

![image](https://github.com/astropy/astropy/assets/311639/e4f7f4bb-85e1-4408-a0d3-55504821c4a2)